File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if target > nums[-1]:
            return len(nums)
        
        i = 0

        while True:
            if nums[i] == target:
                return i
            
            if nums[i] > target:
                return i
            
            i += 1

File: 0035-search-insert-position/0035-search-insert-position.go
func searchInsert(nums []int, target int) int {
    var l int = len(nums)
    if target > nums[l - 1] {
        return l
    }
    
    var idx int = 0
    for {
        if nums[idx] == target {
            break
        }
        
        if nums[idx] > target {
            break
        }
        
        idx += 1
    }
    
    return idx
}